specification/kubernetesconfiguration/resource-manager/readme.md

```diff
@@ -0,0 +1,359 @@
+# kubernetesconfiguration
+
+> see https://aka.ms/autorest
+
+This is the AutoRest configuration file for KubernetesConfiguration.
+
+## Getting Started
+
+To build the SDKs for KubernetesConfiguration, simply install AutoRest via `npm` (`npm install -g autorest`) and then run:
+
+> `autorest readme.md`
+
+To see additional help and options, run:
+
+> `autorest --help`
+
+For other options on installation see [Installing AutoRest](https://aka.ms/autorest/install) on the AutoRest github page.
+
+---
+
+## Configuration
+
+### Basic Information
+
+These are the global settings for the KubernetesConfiguration.
+
+``` yaml
+title: SourceControlConfigurationClient
+description: KubernetesConfiguration Client
+openapi-type: arm
+tag: package-preview-2024-04
+```
+
+### Tag: package-preview-2024-04
+
+These settings apply only when `--tag=package-preview-2024-04` is specified on the command line.
+
+``` yaml $(tag) == 'package-preview-2024-04'
+input-file:
+  - Microsoft.KubernetesConfiguration/stable/2023-05-01/extensions.json
+  - Microsoft.KubernetesConfiguration/preview/2024-04-01-preview/fluxconfiguration.json
+  - Microsoft.KubernetesConfiguration/stable/2023-05-01/kubernetesconfiguration.json
+  - Microsoft.KubernetesConfiguration/stable/2023-05-01/operations.json
+suppressions:  
+  - code: ResourceNameRestriction
+    from: fluxconfiguration.json
+    reason: Existing service contract needs to be backward compatible, pattern validation exists in RP.
```

Comment:
Suppressions should be scoped with filters to match them to the path of the violation, so that you don't get false negatives where one suppression direction suppresses genuine issues by suppressing multiple issues